plugin/rollback.go

```diff
@@ -93,26 +101,46 @@ func (b *backend) serviceAccountKeyRollback(ctx context.Context, req *logical.Re
 		return err
 	}
 
-	b.Logger().Debug("checking roleset listed in WAL is access_token roleset")
-
+	b.Logger().Debug("checking parent listed in WAL generates access_token secret")
 	var keyInUse string
 
-	// Get roleset for entry
-	rs, err := getRoleSet(entry.RoleSet, ctx, req.Storage)
-	if err != nil {
-		return err
-	}
+	switch {
+	case entry.RoleSet != "":
+		rs, err := getRoleSet(entry.RoleSet, ctx, req.Storage)
+		if err != nil {
+			return err
+		}
 
-	// If roleset is not nil, get key in use.
-	if rs != nil {
-		if rs.SecretType != SecretTypeAccessToken {
-			// Don't clean keys if roleset doesn't create access_tokens (i.e. creates keys).
-			return nil
+		// If roleset is not nil, get key in use.
+		if rs != nil {
+			if rs.SecretType != SecretTypeAccessToken {
+				// Remove WAL entry - we don't clean keys if roleset generates key secrets.
+				return nil
+			}
+
+			if rs.TokenGen != nil {
+				keyInUse = rs.TokenGen.KeyName
+			}
+		}
+	case entry.StaticAccount != "":
+		sa, err := b.getStaticAccount(entry.StaticAccount, ctx, req.Storage)
+		if err != nil {
+			return err
 		}
 
-		if rs.TokenGen != nil {
-			keyInUse = rs.TokenGen.KeyName
+		// If roleset is not nil, get key in use.
+		if sa != nil {
+			if sa.SecretType != SecretTypeAccessToken {
+				// Remove WAL entry - we don't clean keys if roleset generates key secrets.
+				return nil
+			}
+			if sa.TokenGen != nil {
+				keyInUse = sa.TokenGen.KeyName
+			}
 		}
+	default:
+		b.Logger().Error("removing invalid walAccountKey with empty RoleSet and empty StaticAccount - may need manual cleanup: %v", entry)
```

Comment:
Nit:
```suggestion
		b.Logger().Error("removing invalid walAccountKey with empty RoleSet and empty StaticAccount, may need manual cleanup: %v", entry)
```

Reply:
I realized that a lot of existing errors uses `-` to break up the message so totally optional if we want to keep things consistent.